src/whisper.cpp

```diff
@@ -5197,7 +5198,7 @@ struct whisper_vad_speech whisper_vad_detect_speech(struct whisper_vad_context *
 
     struct whisper_vad_speech speech = {
         /* n_probs = */ n_chunks,
-        /* probs   = */ probs,
+        /* probs   = */ vctx->state->probs.data(),
```

Comment:
@ggerganov Do you have any suggestion about doing this, that is having this return a pointer to the vad state. Should I copy this instead, or is there some other pattern I should follow, perhaps something similar is done elsewhere in the code base that I'm not aware of?

Reply:
Returning a pointer to the VAD state is correct. This way the VAD state remains the owner of the buffer and the user does not have to explicitly release it.